parlai/core/params.py

```diff
@@ -127,22 +150,24 @@ def add_model_args(self, args=None):
             if hasattr(agent, 'add_cmdline_args'):
                 agent.add_cmdline_args(self)
             if hasattr(agent, 'dictionary_class'):
-                self.add_argument(
-                    '--dict-class', default=agent.dictionary_class(), type=str,
-                    help='the class of the dictionary agent used')
+                model_args.set_defaults(dict_class=agent.dictionary_class())
```

Comment:
yes the whole class--I thought it was better because when you are using this opt, you're likely to be initializing an instance of it, not just inspecting the name, so then you can just do that instead of also having to figure out how to import from a string when you want to use it. (see the change to examples/build_dict.py: that seems much more natural to me)